homeassistant/components/myuplink/sensor.py

```diff
@@ -102,4 +152,36 @@ def __init__(
     def native_value(self) -> StateType:
         """Sensor state value."""
         device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        if device_point.value == -32768:
+            return None
         return device_point.value  # type: ignore[no-any-return]
+
+
+class MyUplinkEnumSensor(MyUplinkDevicePointSensor):
+    """Representation of a myUplink device point sensor for ENUM device_class."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            device_point=device_point,
+            entity_description=entity_description,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        self._attr_options = [x["text"].capitalize() for x in device_point.enum_values]
+        self.options_map = {x["value"]: x["text"] for x in device_point.enum_values}
+
+    @property
+    def native_value(self) -> str:
+        """Sensor state value for enum sensor."""
+        device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        return self.options_map[str(int(device_point.value))].capitalize()  # type: ignore[no-any-return]
```

Comment:
I think the enum value should be the raw value (or a stringified version of that). The value should not be language dependant.

Reply:
I dont really follow your point.
The raw values for a typical enum sensor are [0,1,2,3] and meaningless for displaying in the UI for the normal user. This implementation extracts the text mappings from the device_point data so we can display meaningful text to the user e.g. [Running, Starting, Blocked]. An alternative could be to map numbers to slugs and use state translation in Lokalise/HA. But that will put a huge maintenance burden on the community. These enum texts are dynamic from the API and the translation_keys in HA are static by nature. A more simple approach is to store the raw value as an attribute to the sensor.

Reply:
I meant to slugs yes. I know translations would be tricky. I dont like that the value is language dependant though. If it was fixed english it would be okey. Breaking automations due to changing HA language is not good.

Can you split the enum stuff from this pull. Im good with the rest. Then we can look at that separate.

Reply:
I am sort of reluctant to remove the enum stuff because it good for the non-tech nerds.
Note that all strings are in english in the integration today so there is no real problem until the language switching PR is scrutinized.

I have two alternate proposals that will make it possible to change system language in HA without reviewing/updating automations: 

1. Add an extra_state_attribute to the enum sensors with the raw value. That attribute can be used for automations by people that switches system language now and then.
2. Add duplicate sensors for the enum raw values with the sensor name suffixed by "_raw". These sensors can be hidden from the UI or disabled by default.

Any opinions on this? @MartinHjelmare @pajzo 

Regarding:
> That constant kind of needs to be defined in the library, but it at least need a constant defined for it here.

I can change to a constant e.g. (MARKER_FOR_UNKNOWN_VALUE) here in this module for now. I will move the check to the lib in next release. OK?

Reply:
Ok